include/zephyr/dt-bindings/input/keymap.h

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_INPUT_KEYMAP_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_INPUT_KEYMAP_H_
+
+#define MATRIX_KEY(row, col, code) \
+	((((row) & 0xff) << 24) | (((col) & 0xff) << 16) | ((code) & 0xffff))
+
+#define MATRIX_ROW(keymap_entry) (((keymap_entry) >> 24) & 0xff)
+#define MATRIX_COL(keymap_entry) (((keymap_entry) >> 16) & 0xff)
```

Comment:
if these are not meant for DT usage, they should reside somewhere else

Reply:
yeah good point, I forsee using these somewhere else in the future, moved them in `include/input`